### ISSUE TITLE
Can the methods in causalml be trained with incremental training?

### ISSUE THREAD

**Issue by evelynjin990** [uber/causalml#622]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by jeongyoonlee** [comment]
Hi — thanks for raising this.

CausalML is model-agnostic: it wraps any scikit-learn-compatible estimator. For incremental training, you can pass a base learner that supports `partial_fit` (e.g., `SGDRegressor`, `MiniBatchKMeans`) as the underlying model. For example:

```python
from sklearn.linear_model import SGDRegressor
from causalml.inference.meta import BaseSLearner

learner = BaseSLearner(learner=SGDRegressor())
```

That said, the meta-learner `fit()` method itself calls `model.fit()` (not `partial_fit()`), so you'd need to handle the incremental loop externally — fitting the base models in batches and then using the meta-learner for prediction.

For tree-based methods like `CausalRandomForestRegressor`, incremental training isn't natively supported by the underlying algorithms (scikit-learn random forests don't support `partial_fit` either). For large datasets, consider subsampling or using `warm_start=True` with increasing `n_estimators`.

We don't currently plan to add incremental fitting infrastructure to CausalML itself, but contributions that make this pattern easier are welcome.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — I think this one is essentially answered and can be closed.

`causalml`'s meta-learners are model-agnostic: anything wrapping a scikit-learn-compatible estimator with `partial_fit` (e.g. `SGDRegressor`, `MultinomialNB`) gets you incremental training out of the box, since the meta-learner just calls `model.fit(...)` on its inner learner. See e.g. `causalml/inference/meta/base.py` for the wrapper pattern (`fit(X, treatment, y, p=None)`).

Adding a first-class incremental-training mode at the meta-learner level (e.g. a stable `partial_fit`) would be a separate, larger feature — worth a fresh issue with a concrete API sketch if anyone wants to take it on.

(Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually against the current `master` branch.)
